purchase_order_import/wizard/order_response_import.py

```diff
@@ -297,100 +294,132 @@ def _process_conditional(self, purchase_order, parsed_order_document):
+                and float_compare(
+                    qty, move.product_qty, precision_digits=precision
+                )
+                < 0
+            ):
+                # qty planned < qty into the stock move: Split it
+                new_move_id = move.split(move.product_qty - qty)
+                move = self.env["stock.move"].browse(new_move_id)
+            qty -= move.product_qty
+            if not backorder_qty:
+                # if no backorder -> we must cancel the move
+                move_ids_to_cancel.append(move.id)
+                continue
+            # from here we process the backorder qty
+            # we distribute this qty into the remaining moves and
+            # if this qty is < than the expected one, we split and cancel the
+            # remaining qty
+            if (
+                float_compare(
+                    backorder_qty, move.product_qty, precision_digits=precision
+                )
+                < 0
+            ):
+                # backorder_qty < qty into the move -> split the move
+                # anf cancel remaining qty
+                move_ids_to_cancel.append(
+                    move.split(move.product_qty - backorder_qty)
+                )
+            if move.split_from:
+                # Add not on the original move to say that items will be
+                # delivered into a next delivery
+                note = line_info.get("note")
+                note = note + "\n" if note else ""
+                note += (
+                    _("%s items should be delivered into a next delivery.")
+                    % move.product_qty
+                )
+                move.split_from.note = note
+            backorder_qty -= move.product_qty
+            move_ids_to_backorder.append(move.id)
+        # move backorder moves to a backorder
+        if move_ids_to_backorder:
+            moves_to_backorder = self.env["stock.move"].browse(
+                move_ids_to_backorder
+            )
+            self._add_moves_to_backorder(moves_to_backorder)
+        # cancel moves to cancel
+        if move_ids_to_cancel:
+            moves_to_cancel = self.env["stock.move"].browse(move_ids_to_cancel)
+            moves_to_cancel.action_cancel()
+            moves_to_cancel.write(
+                {"note": _("No backorder planned by the supplier.")}
+            )
+        # Reset Operations
+        moves[0].picking_id.do_prepare_partial()
 
     @api.model
-    def _add_move_to_backorder(self, move):
+    def _add_moves_to_backorder(self, moves):
         """
-        Add the move the picking's backorder
+        Add the moves to the picking's backorder
         return the backorder associated to the current picking. If no backorder
         exists, create a new one.
-        :param move:
+        :param moves:
         """
         StockPicking = self.env["stock.picking"]
-        current_picking = move.picking_id
+        current_picking = moves[0].picking_id
```

Comment:
Perhaps it is safer to group the moves by picking and iterate over the combinations?